zerver/tests/tests.py

```diff
@@ -586,6 +601,47 @@ def test_add_bot(self):
         bot = bots[0]
         self.assertEqual(bot['bot_owner'], 'hamlet@zulip.com')
 
+    def test_add_bot_with_username_in_use(self):
+        # type: () -> None
+        self.login("hamlet@zulip.com")
+        self.assert_num_bots_equal(0)
+        result = self.create_bot()
+        self.assert_num_bots_equal(1)
+
+        bot_info = dict(
+            full_name='whatever',
```

Comment:
stylistically I might make the full_name be "Duplicate" to make the code a bit more readable, but this is fine :)